### ISSUE TITLE
bootstrap.sh is broken on Linux with GCC 6 (for example, on Debian 9)

### ISSUE THREAD

**Issue by kamahall** [microsoft/vcpkg#17897]
**Describe the bug**

```
-- Build files have been written to: /root/vcpkg/buildtrees/_vcpkg/build
[0/2] Re-checking globbed directories...
[4/91] Building CXX object CMakeFiles/vcpkg.dir/src/vcpkg.cpp.o
FAILED: CMakeFiles/vcpkg.dir/src/vcpkg.cpp.o
/usr/bin/g++-6 -DVCPKG_BASE_VERSION=2021-05-05 -DVCPKG_USE_STD_FILESYSTEM=0 -DVCPKG_VERSION=unknownhash -I/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include -O3 -DNDEBUG -std=c++1z -MD -MT CMakeFiles/vcpkg.dir/src/vcpkg.cpp.o -MF CMakeFiles/vcpkg.dir/src/vcpkg.cpp.o.d -o CMakeFiles/vcpkg.dir/src/vcpkg.cpp.o -c /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp: In function ‘void load_config(vcpkg::Files::Filesystem&)’:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:123:55: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>&&)’
         auto locked_metrics = Metrics::g_metrics.lock();
                                                       ^
In file included from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:22:12: note: ‘vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>&&)’ is implicitly deleted because the default definition would be ill-formed:
     struct LockGuardPtr
            ^~~~~~~~~~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:22:12: error: use of deleted function ‘std::lock_guard<_Mutex>::lock_guard(const std::lock_guard<_Mutex>&) [with _Mutex = std::mutex]’
In file included from /usr/include/c++/6/mutex:44:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:5,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/usr/include/c++/6/bits/std_mutex.h:170:7: note: declared here
       lock_guard(const lock_guard&) = delete;
       ^~~~~~~~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp: In lambda function:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:186:48: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>&&)’
         auto metrics = Metrics::g_metrics.lock();
                                                ^
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp: In function ‘int main(int, const char* const*)’:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:207:55: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>&&)’
         auto locked_metrics = Metrics::g_metrics.lock();
                                                       ^
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:235:48: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>&&)’
         auto metrics = Metrics::g_metrics.lock();
                                                ^
In file included from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h: In instantiation of ‘vcpkg::Util::LockGuardPtr<T> vcpkg::Util::LockGuarded<T>::lock() [with T = vcpkg::Metrics::Metrics]’:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:51:29:   required from here
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:14:42: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Metrics::Metrics>&&)’
         LockGuardPtr<T> lock() { return *this; }
                                          ^~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:29:9: note:   after user-defined conversion: vcpkg::Util::LockGuardPtr<T>::LockGuardPtr(vcpkg::Util::LockGuarded<T>&) [with T = vcpkg::Metrics::Metrics]
         LockGuardPtr(LockGuarded<T>& sync) : m_lock(sync.m_mutex), m_ptr(sync.m_t) { }
         ^~~~~~~~~~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h: In instantiation of ‘vcpkg::Util::LockGuardPtr<T> vcpkg::Util::LockGuarded<T>::lock() [with T = std::__cxx11::basic_string<char>]’:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:137:36:   required from here
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:14:42: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<std::__cxx11::basic_string<char> >::LockGuardPtr(vcpkg::Util::LockGuardPtr<std::__cxx11::basic_string<char> >&&)’
         LockGuardPtr<T> lock() { return *this; }
                                          ^~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:22:12: note: ‘vcpkg::Util::LockGuardPtr<std::__cxx11::basic_string<char> >::LockGuardPtr(vcpkg::Util::LockGuardPtr<std::__cxx11::basic_string<char> >&&)’ is implicitly deleted because the default definition would be ill-formed:
     struct LockGuardPtr
            ^~~~~~~~~~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:22:12: error: use of deleted function ‘std::lock_guard<_Mutex>::lock_guard(const std::lock_guard<_Mutex>&) [with _Mutex = std::mutex]’
In file included from /usr/include/c++/6/mutex:44:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:5,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/usr/include/c++/6/bits/std_mutex.h:170:7: note: declared here
       lock_guard(const lock_guard&) = delete;
       ^~~~~~~~~~
In file included from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:29:9: note:   after user-defined conversion: vcpkg::Util::LockGuardPtr<T>::LockGuardPtr(vcpkg::Util::LockGuarded<T>&) [with T = std::__cxx11::basic_string<char>]
         LockGuardPtr(LockGuarded<T>& sync) : m_lock(sync.m_mutex), m_ptr(sync.m_t) { }
         ^~~~~~~~~~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h: In instantiation of ‘vcpkg::Util::LockGuardPtr<T> vcpkg::Util::LockGuarded<T>::lock() [with T = vcpkg::Chrono::ElapsedTimer]’:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:167:30:   required from here
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:14:42: error: use of deleted function ‘vcpkg::Util::LockGuardPtr<vcpkg::Chrono::ElapsedTimer>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Chrono::ElapsedTimer>&&)’
         LockGuardPtr<T> lock() { return *this; }
                                          ^~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:22:12: note: ‘vcpkg::Util::LockGuardPtr<vcpkg::Chrono::ElapsedTimer>::LockGuardPtr(vcpkg::Util::LockGuardPtr<vcpkg::Chrono::ElapsedTimer>&&)’ is implicitly deleted because the default definition would be ill-formed:
     struct LockGuardPtr
            ^~~~~~~~~~~~
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:22:12: error: use of deleted function ‘std::lock_guard<_Mutex>::lock_guard(const std::lock_guard<_Mutex>&) [with _Mutex = std::mutex]’
In file included from /usr/include/c++/6/mutex:44:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:5,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/usr/include/c++/6/bits/std_mutex.h:170:7: note: declared here
       lock_guard(const lock_guard&) = delete;
       ^~~~~~~~~~
In file included from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/metrics.h:4:0,
                 from /root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/src/vcpkg.cpp:17:
/root/vcpkg/buildtrees/_vcpkg/src/vcpkg-tool-2021-05-05-9f849c4c43e50d1b16186ae76681c27b0c1be9d9/include/vcpkg/base/lockguarded.h:29:9: note:   after user-defined conversion: vcpkg::Util::LockGuardPtr<T>::LockGuardPtr(vcpkg::Util::LockGuarded<T>&) [with T = vcpkg::Chrono::ElapsedTimer]
         LockGuardPtr(LockGuarded<T>& sync) : m_lock(sync.m_mutex), m_ptr(sync.m_t) { }
         ^~~~~~~~~~~~
[9/91] Building CXX object CMakeFiles/vcpkglib.dir/src/vcpkg/base/files.cpp.o
ninja: build stopped: subcommand failed.
```

**Environment**
- OS: [e.g. Windows/Linux etc...]
- Compiler: revision

**To Reproduce**
1. ./vcpkg install xxxx
2. See error
Repro code when

**Expected behavior**

**Failure logs** 
-(please attached failure logs)

**Additional context**

### I am installing vcpkg on Ubuntu Debian 9 and doing the command

./vcpkg/bootstrap-vcpkg.sh

it gives me this error

note: so download it in debian 9 `git clone https://github.com/microsoft/vcpkg`

**Comment by dg0yt** [comment]
`/usr/bin/g++-6` is too old. Compiling vcpkg-tool needs C++17.

**Comment by kamahall** [comment]
> `/usr/bin/g++-6` is too old. Compiling vcpkg-tool needs C++17.

Can you help me fix it, can you give me your discord?

**Comment by JackBoosY** [comment]
1. Please make sure your vcpkg is latest.
2. Please update gcc version, see [documentation](https://linuxize.com/post/how-to-install-gcc-compiler-on-debian-10/).

**Comment by JackBoosY** [comment]
We haven't been able to repro this; if more information comes up, or this issue appears again, please reopen.

**Comment by talregev** [comment]
It happen to me. 
I am using fdroid docker image file. I guess the gcc there is old.
How I install a new one? 
Can we open the issue?

**Comment by talregev** [comment]
I found how to install g++ 9 on Debian:
I am trying this answer to see if it help. 
https://stackoverflow.com/questions/61590804/how-to-install-gcc-9-on-debianlatest-docker

**Comment by talregev** [comment]
Also this for `>>`
https://stackoverflow.com/questions/11301650/how-to-escape-indicator-characters-i-e-or-in-yaml

**Comment by talregev** [comment]
Didn't help. from here alternative to append to file: 
https://stackoverflow.com/questions/6207573/how-to-append-output-to-the-end-of-a-text-file

**Comment by talregev** [comment]
In fdroid docker image when installing g++, it install g++ v6. 
I try to modify it without success. If someone know how to do it, that will be great.

**Comment by talregev** [comment]
I found a way to compile vcpkg inside Debian docker image inside fdroid ci. 
Install clang-7
https://gitlab.com/fdroid/fdroiddata/-/blob/master/metadata/org.musicpd.yml#L298

**Comment by talregev** [comment]
@JackBoosY 
But I have a problem on a post build in openssl android, with clean. It try to delete `armv6-android.zip` and fail the ci.
I think this ticket can be close and we need to open a new one. 
```
-- Performing post-build validation
-- Performing post-build validation done
remove(/home/vagrant/build/edu.berkeley.boinc/3rdParty/android/vcpkg/buildtrees/openssl/armv6-android.zip): No such file or directory 
```

**Comment by talregev** [comment]
Maybe the Debian image is missing the `zip` package?

**Comment by JackBoosY** [comment]
@talregev Can you provide the original error info?

**Comment by talregev** [comment]
@JackBoosY Can I send you a link to gitlab ci?

**Comment by talregev** [comment]
https://gitlab.com/tal.regev/fdroiddata/-/jobs/1475472268

**Comment by JackBoosY** [comment]
@talregev That should be a port bug instead of a vcpkg bug, can you please open a new issue to report that?

**Comment by talregev** [comment]
It not port bug, because when we compile the same on ubuntu it not happen. 
It vcpkg on different os bug. (maybe also I compile it with clang?).

**Comment by JackBoosY** [comment]
In any case, the error log you provided is completely inconsistent with the original error information of this issue, so please open a new issue.

**Comment by talregev** [comment]
I agree. I already suggest how to solve it. 
You should update your documentation. 
https://github.com/microsoft/vcpkg#quick-start-unix
It should g++ > 9
also clang > 5 or 7. (not sure for the c++17 inside it).

**Comment by JackBoosY** [comment]
> I agree. I already suggest how to solve it.
> You should update your documentation.
> https://github.com/microsoft/vcpkg#quick-start-unix
> It should g++ >= 9
> or clang++ >= 5 or 7. (not sure for the c++17 inside it).
> @JackBoosY Will you update documentation? Maybe add a clang to linux section?
> Also add that you are looking for c++17.

I think we should fix using gcc to build vcpkg, not clang.
cc @BillyONeal

**Comment by talregev** [comment]
You can change to gcc > 9 for ubuntu
and debian gcc 9 and 10 are still in testing. So other option for them is clang. 
What do you say?

**Comment by talregev** [comment]
@JackBoosY Please remain this issue open until you modify the readme.

**Comment by talregev** [comment]
> In any case, the error log you provided is completely inconsistent with the original error information of this issue, so please open a new issue.

I was thinking about open a ticket.      
But I can only re-confirm the bug only on gitlab ci on fdroid.    
I try to do it local on my wsl Debian. They using Debian 9, wsl is Debian 10, and on the wsl is not happening.      
Also, it will be hard for me to provide a necessary logs, and they also are going to switch to Debian 10 somewhere in the future.       
@JackBoosY What do you think?

**Comment by JackBoosY** [comment]
@talregev  If it happens again, we should keep this PR open until we fix it.

**Comment by talregev** [comment]
> @talregev If it happens again, we should keep this PR open until we fix it.

There are 2 bugs we talking here.
1. Ticket bug. that always happen, and the solution is to modify readme, and explain we need gcc > 7 with c++17.
2. The mysterious remove zip file that happen on openssl android ports (maybe on other ports too).
It always happen in the fdroid ci. Didn't happen on my local wsl.

**Comment by talregev** [comment]
> > In any case, the error log you provided is completely inconsistent with the original error information of this issue, so please open a new issue.
> 
> I was thinking about open a ticket.
> But I can only re-confirm the bug only on gitlab ci on fdroid.
> I try to do it local on my wsl Debian. They using Debian 9, wsl is Debian 10, and on the wsl is not happening.
> Also, it will be hard for me to provide a necessary logs, and they also are going to switch to Debian 10 somewhere in the future.
> @JackBoosY What do you think?

I was talking about bug number 2 here.

**Comment by JackBoosY** [comment]
```
remove(/home/vagrant/build/edu.berkeley.boinc/3rdParty/android/vcpkg/buildtrees/openssl/armv6-android.zip): No such file or directory
```
For this, I really don't know who did this.

**Comment by talregev** [comment]
Can you take a look on openssl port and tell me where is the line that create `armv6-android.zip` then remove?

**Comment by talregev** [comment]
Also I notice there is an `openssl-unix`. For what reason there is this port?

**Comment by talregev** [comment]
This error also happen in different port `expat:armv6-android`. My guess is that is something connected to this image. and nor from vcpkg.
Just update the documantion:
gcc >=7 with c++17
https://gcc.gnu.org/projects/cxx-status.html#cxx17

**Comment by JackBoosY** [comment]
Can you please provide the file list when unzip `armv6-android.zip` ?

**Comment by talregev** [comment]
Not sure I can. and I think this zip is not exist anyway. (it not find it). 
I will try to output the log of this weird case. 
(it not my repo, and it CI, so make a little harder to make things).
I will try to do it next week and give you a full report from what I learn.

**Comment by JackBoosY** [comment]
@talregev Also, please open a new issue to report that.

**Comment by talregev** [comment]
ok. I will do it next week.

**Comment by BillyONeal** [comment]
> You can change to g++ > 9 for ubuntu
> and debian g++ 9 and g ++ 10 are still in testing. So other option for them is clang.
> What do you say?

We added CI testing that it should work all the way back to GCC 7. https://github.com/microsoft/vcpkg-tool/pull/122/files#diff-cc214df161ca5fc74d96213dce0e5c58201f97b8aad43a451f5c4e27c7cd8a30R43

>But I can only re-confirm the bug only on gitlab ci on fdroid.

To clarify, is the problem that:

* They have a recent enough version of gcc to build vcpkg, but not a recent enough version of clang to build vcpkg.
* bootstrap.sh tries clang before gcc, and so chooses the clang that can't build vcpkg.

I would like to add testing for whatever configuration is causing the problem

**Comment by BillyONeal** [comment]
Ah wait I reread this and understand now, the problem is that for some reason gcc-6 is being selected, but the testing we added was for 7.

**Comment by talregev** [comment]
@BillyONeal 
Do you want support gcc 6 with vcpkg?

**Comment by talregev** [comment]
> Ah wait I reread this and understand now, the problem is that for some reason gcc-6 is being selected, but the testing we added was for 7.

In the system by default there is no gcc on Debian. if you install build-essential or gcc or g++ it will install gcc 6.
In gcc 6 there isn't full c++17 needed for compile vcpkg.
Are you want to remove these compilation from vcpkg to support gcc 6 or you want to update documentation that you support gcc 7 and above?

**Comment by BillyONeal** [comment]
> @BillyONeal
> Do you want to support gcc 6 with vcpkg?

That's what our readme says and we didn't consciously decide to drop support for it, so we want it to work, hence my PR fixing it and adding testing for it. https://github.com/microsoft/vcpkg-tool/pull/150

>In the system by default there is no gcc on Debian. if you install build-essential or gcc or g++ it will install gcc 6.

That's what I mean by "default compiler" -- if you install a compiler out of their repos, you get gcc6 unless you jump through hoops.

>In gcc 6 there isn't full implementation of c++17 needed for compile vcpkg.

We don't need full '17, we need std::filesystem or std::experimental::filesystem: I have been actively working on a project to drop the filesystem dependency so we can go to even older Linuxes.

**Comment by talregev** [comment]
Ok. I will wait for your PR. Can you mention this issue that your PR will fix? Thank you.

**Comment by BillyONeal** [comment]
> Can you mention on your PR that it will fix this issue?

It already says that

**Comment by talregev** [comment]
If you type on your PR (the first msg) Fix #17897. It also mark your PR as fixing the PR. also show in the right.

**Comment by talregev** [comment]
Example from this PR:
https://github.com/microsoft/vcpkg/pull/19461
![image](https://user-images.githubusercontent.com/7043539/128767673-eed2cdd2-c0a7-4504-beac-6b62e321234a.png)

**Comment by BillyONeal** [comment]
That doesn't work when the fix is in a different repo

**Comment by talregev** [comment]
ok.

**Comment by BillyONeal** [comment]
> ok. So maybe we can ask Microsoft a feature for that ...

I'm pretty sure it's intentional, to prevent accidentally modifying one repo when making a change to another repo.

**Comment by talregev** [comment]
ok. But Usually when you put a link to different repo, it show you and you know how it look like. you don't just give a number.
I think it will be helpful as the current Fix mention.
And thank you for the information.

**Comment by BillyONeal** [comment]
Reopening because this needs a tool release.

**Comment by BillyONeal** [comment]
Apparently it does close across repos if the author has permissions O_O

**Comment by talregev** [comment]
@BillyONeal How we should know when a new vcpkg-tool will come in?

**Comment by linquize** [comment]
Visit the repository https://github.com/Microsoft/vcpkg-tool

**Comment by linquize** [comment]
@talregev yes

**Comment by BillyONeal** [comment]
I'll do a tool release today to get this and to unblock @Neumann-A 's ifort stuff

**Comment by talregev** [comment]
@BillyONeal Thank you!
